Mathlib/Data/Matroid/Closure.lean

```diff
@@ -753,5 +849,123 @@ lemma ext_spanning {M M' : Matroid α} (h : M.E = M'.E)
+
+@[simp] lemma loopyOn_spanning_iff {E : Set α} : (loopyOn E).Spanning X ↔ X ⊆ E := by
+  rw [spanning_iff, loopyOn_closure_eq, loopyOn_ground, and_iff_right rfl]
+
+lemma closure_empty_eq_ground_iff : M.closure ∅ = M.E ↔ M = loopyOn M.E := by
+  refine ⟨fun h ↦ ext_closure ?_, fun h ↦ by rw [h, loopyOn_closure_eq, loopyOn_ground]⟩
+  refine fun X ↦ subset_antisymm (by simp [closure_subset_ground]) ?_
+  rw [loopyOn_closure_eq, ← h]
+  exact M.closure_mono (empty_subset _)
+
+@[simp] lemma uniqueBaseOn_closure_eq (I E X : Set α) :
+    (uniqueBaseOn I E).closure X = (X ∩ I ∩ E) ∪ (E \ I) := by
+  have hb : (uniqueBaseOn (I ∩ E) E).Basis (X ∩ E ∩ (I ∩ E)) (X ∩ E) :=
+    (uniqueBaseOn_basis_iff inter_subset_right).2 rfl
+  ext e
+  rw [← uniqueBaseOn_inter_ground_eq I E, ← closure_inter_ground _ X, uniqueBaseOn_ground,
+    ← hb.closure_eq_closure, hb.indep.mem_closure_iff, dep_iff, uniqueBaseOn_indep_iff',
+    insert_subset_iff, uniqueBaseOn_ground, inter_assoc, inter_self,
+    and_iff_left inter_subset_right, ← inter_inter_distrib_right, inter_assoc,
+    inter_union_distrib_right, inter_comm I, inter_union_diff, insert_subset_iff, inter_comm X,
+    inter_assoc, and_iff_left inter_subset_left, mem_inter_iff]
+  simp only [not_and, mem_inter_iff, mem_union, mem_diff]
+  tauto
+
+@[simp] lemma restrict_closure_eq' (M : Matroid α) (X R : Set α) :
+    (M ↾ R).closure X = (M.closure (X ∩ R) ∩ R) ∪ (R \ M.E) := by
+  rw [← closure_inter_ground, restrict_ground_eq]
+  ext e
+  obtain ⟨I, hI⟩ := (M ↾ R).exists_basis (X ∩ R)
+  have hI' := (basis_restrict_iff'.mp hI).1
+  rw [← hI.closure_eq_closure, ← M.closure_inter_ground (X ∩ R), ← hI'.closure_eq_closure,
+    mem_union, mem_inter_iff, hI'.indep.mem_closure_iff, hI.indep.mem_closure_iff, restrict_dep_iff,
+    insert_subset_iff, dep_iff, insert_subset_iff, and_iff_left hI'.indep.subset_ground, mem_diff,
+    and_iff_left (show I ⊆ R from hI.indep.subset_ground)]
+  have hIR : I ⊆ R := hI.indep.subset_ground
+  by_cases he : e ∈ M.E
+  · aesop
+  simp only [iff_false_intro he, and_false, false_or, and_true, ← mem_inter_iff, ← mem_union,
+    inter_eq_self_of_subset_left hIR, union_comm I, and_iff_right
+      (show ¬M.Indep (insert e I) from fun hi ↦ he (hi.subset_ground (mem_insert _ _))),
+    not_false_iff]
+
+lemma restrict_closure_eq (M : Matroid α) (hXR : X ⊆ R) (hR : R ⊆ M.E := by aesop_mat) :
+    (M ↾ R).closure X = M.closure X ∩ R := by
+  rw [restrict_closure_eq', diff_eq_empty.mpr hR, union_empty, inter_eq_self_of_subset_left hXR]
+
+@[simp] lemma comap_closure_eq {β : Type*} (M : Matroid β) (f : α → β) (X : Set α) :
+    (M.comap f).closure X = f ⁻¹' M.closure (f '' X) := by
+  obtain ⟨I, hI⟩ := (M.comap f).exists_basis' X
+  obtain ⟨hI', hIinj, -⟩ := comap_basis'_iff.1 hI
+  rw [← hI.closure_eq_closure]
+  ext x
+  obtain (hxE | hxE) := em' (f x ∈ M.E)
+  · apply iff_of_false <;> exact (fun h ↦ hxE (by simpa using mem_ground_of_mem_closure h))
+
```

Comment:
I'm guessing you are adding empty lines to indicate a "new step" in the proof?
Please use the empty line as an opportunity to add a short comment instead.

Similarly below.